universal_transfer_operator/src/universal_transfer_operator/integrations/fivetran.py

```diff
@@ -0,0 +1,292 @@
+
+        if group_id is None:
+            logging.warning(
+                "group_id is None. It should be the unique identifier for "
+                "the group within the Fivetran system. "
+            )
+            return False
+        endpoint = self.api_path_groups + group_id
+        api_response = fivetran_hook._do_api_call(("GET", endpoint))
+        if api_response["code"] == "Success":
+            logging.info("group_id {group_id} found.", extra={"group_id": group_id})
+        else:
+            raise ValueError(api_response)
+        return True
+
+    def create_group(self, fivetran_hook: FivetranHook) -> str:
+        """
+        Creates the group based on group name passed
+        """
+        endpoint = self.api_path_groups
+        group_dict = self.transfer_params.get("group", None)
+        if group_dict is None:
+            raise ValueError("Group is none. Pass a valid group")
+        group = Group(**group_dict)
+        payload = {"name": group.name}
+        api_response = fivetran_hook._do_api_call(("POST", endpoint), json=payload)
+        if api_response["code"] == "Success":
+            logging.info(api_response)
+        else:
+            raise ValueError(api_response)
+        return api_response["data"]["id"]
+
+    def check_destination_details(self, fivetran_hook: FivetranHook, destination_id: str | None) -> bool:
+        """
+        Check if destination_id is exists.
+        """
+        if destination_id is None:
+            logging.warning(
+                "destination_id is None. It should be the unique identifier for "
+                "the destination within the Fivetran system. "
+            )
+            return False
+        endpoint = self.api_path_destinations + destination_id
+        api_response = fivetran_hook._do_api_call(("GET", endpoint))
+        if api_response["code"] == "Success":
+            logging.info("destination_id {destination_id} found.", extra={"destination_id": destination_id})
+        else:
+            raise ValueError(api_response)
+        return True
+
+    def create_destination(self, fivetran_hook: FivetranHook, group_id: str) -> dict:
+        """
+        Creates the destination based on destination configuration passed
+        """
+        endpoint = self.api_path_destinations
+        destination_dict = self.transfer_params.get("destination", None)
+        if destination_dict is None:
+            raise ValueError("destination is none. Pass a valid destination")
+        destination = Destination(**destination_dict)
+        payload = {
+            "group_id": group_id,
+            "service": destination.service,
+            "region": destination.region if destination.region else "GCP_US_EAST4",
+            "time_zone_offset": destination.time_zone_offset if destination.time_zone_offset else "-5",
+            "config": destination.config,
+            "run_setup_tests": destination.run_setup_tests if destination.run_setup_tests else True,
```

Comment:
Can we set the defaults in the class definition?

Also why is the default `GCP_US_EAST4` if we don't know the destination will be gcp?

Reply:
I have added the default locations to class definitions.

In this case the `region` here is the data processing location. This is where Fivetran will operate and run computation on data. More details can be found here: https://fivetran.com/docs/rest-api/destinations

Reply:
[Done](https://github.com/astronomer/astro-sdk/pull/1492/commits/8943d29df9d6d565ce668ceb9c7497aa3775c959)

Reply:
Thanks for the explanation :)